### ISSUE TITLE
Add wp_kses_one_attr() to $escapingFunctions

### ISSUE THREAD

**Issue by anastis** [WordPress/WordPress-Coding-Standards#1765]
## Bug Description
`wp_kses_one_attr()` should satisfy escaping rules.

## Minimal Code Snippet
```php
<?php $target = 'target="_blank"'; ?>
<a href="#" <?php echo wp_kses_one_attr( $target, 'a' ); ?>>Link</a>
```

## Error Code
WordPress.Security.EscapeOutput.OutputNotEscaped should be cleared, but it's not.

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.2.1
| PHP_CodeSniffer version | 3.4.0
| WPCS version            | develop
| WPCS install type       | git clone
| IDE (if relevant)       | PhpStorm 2019.1

## Tested Against `develop` branch?
- [y] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by dingo-d** [comment]
Should we update the `$escapingFunctions` list with all the `wp_kses_*` functions? 

The list is here: https://developer.wordpress.org/?s=wp_kses_&post_type%5B%5D=wp-parser-function


@GaryJones @jrfnl ?

**Comment by jrfnl** [comment]
@anastis Thank you for bringing this to our attention.

The problem with functions like `wp_kses_one_attr()` is that they only escape correctly in very specific circumstances.

Before adding it to the `$escapingFunctions` list, I'd like to see the behaviour of this function verified when, for instance, used with the content of a complete post or some other larger snippets of text containing typical attack vectors, to make sure that the function is defensive enough and that the return value is actually safe in all circumstances.

> Should we update the $escapingFunctions list with all the wp_kses_* functions?

@dingo-d No. There are quite a few functions in that list which only provide _partial_ escaping and of which the output should not be considered safe.

**Comment by anastis** [comment]
I'd love to help out, although I'm not quite sure on what you'd like to see tested in terms of the typical attack vectors you mentioned.

I've gone through the function and I believe it's generally safe, due to the way it works.

Maybe you'll want to stop reading at this point, in case you aren't interested in my tests (or they are irrelevant), that hopefully will help convince you `wp_kses_one_attr()` can be added in the list of functions :)

### tl;dr
Input string is split to two parts, attribute and value. If the attribute part isn't whitelisted, an empty string is returned, therefore safe. The value part is either empty or fed through `es_attr()`, again, safe.

### Long version
So,

assuming a call `wp_kses_one_attr( $string, $tag )`, the following inputs behave as such:
`$tag = 'an-invalid-tag';`
Any tag not included in the whitelist array returned (internally) by `wp_kses_allowed_html( 'post' )` will usually  ($string must at least look like an attribute) trigger an undefined index notice, due to unconditional access in `wp_kses_attr_check()`, and an empty string is returned.

So from now on, a valid tag is assumed, e.g. `$tag = 'a';` and it's whitelisted array of attributes `wp_kses_allowed_html( 'post' )['a']`

`$string = 'arbitrary-attribute="value"';`
The attribute name is checked verbatim (converted to lowercase) against the whitelisted array. An empty string is returned.

As  above, `$string = '<script>alert( "oops!" );</script>="value"';` the attribute name is again rejected.

`$string = '<script>alert( "oops!" );</script>';`
Any string that doesn't include a `=` is considered a valueless attribute. Again, it is checked verbatim (lowercase) against the whitelisted array. and an empty string is returned.

In fact, `wp_kses_one_attr()` splits the input string at the first `=` sign. Everything on the left is the attribute, and everything on the right is the value.

Consider some arbitrary html:
`$string = '<a href="#">Link</a>';` results in the following matches:
```
$attribute = '<a href';
$value = '"#">Link</a>';
```
The above will be rejected for two reasons. a) The attribute is not whitelisted. b) The value doesn't end with a single or double quote character.

Expanding on b), any value that doesn't end in a quote character is discarded and an empty string is returned. If it does end in quote character however, it's value is fed to `esc_attr()`, which is known to be a safe escaping function.
E.g. passing `$string = 'href="<script>alert( \'oops!\' );</script>"';`
returns:
`href="&lt;script&gt;alert( &#039;oops!&#039; );&lt;/script&gt;"`
